src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -105,11 +103,67 @@ private async Task PromoteBuildAsync(int PromoteToMaestroChannelId)
                 $"\"SDLValidationContinueOnError\": \"{ _options.SDLValidationContinueOnError }\", " +
                 $"}}";
 
-            await azdoClient.StartNewBuildAsync(BuildPromotionPipelineAccountName, 
+            var azdoBuildId = await azdoClient.StartNewBuildAsync(BuildPromotionPipelineAccountName, 
                 BuildPromotionPipelineProjectName, 
                 BuildPromotionPipelineId, 
+                arcadeSDKSourceBranch,
+                arcadeSDKSourceSHA,
                 queueTimeVariables)
                 .ConfigureAwait(false);
+
+            Console.WriteLine($"Build {build.Id} will be assigned to channel '{targetChannel.Name}' once this promotion build finishes: " +
+                $"https://{BuildPromotionPipelineAccountName}.visualstudio.com/{BuildPromotionPipelineProjectName}/_build/results?buildId={azdoBuildId}&view=results");
+        }
+
+        /// <summary>
+        /// By default the source branch/SHA for the Build Promotion pipeline will be the branch/SHA
+        /// that produced the Arcade.SDK used by the build being promoted. The user can override that
+        /// by specifying both, channel & SHA, on the command line.
+        /// </summary>
+        /// <param name="build">Build for which the Arcade SDK dependency build will be inferred.</param>
+        private async Task<(string sourceBranch, string sourceVersion)> GetSourceBranchInfoAsync(Build build)
+        {
+            if (!string.IsNullOrEmpty(_options.SourceBranch) && !string.IsNullOrEmpty(_options.SourceSHA))
+            {
+                return (_options.SourceBranch, _options.SourceSHA);
+            }
+
+            IGitRepo targetBuildRepoClient = string.IsNullOrEmpty(build.GitHubRepository) ?
+                (IGitRepo) new AzureDevOpsClient(_options.GitLocation, _options.AzureDevOpsPat, Logger, temporaryRepositoryPath: null) :
+                (IGitRepo) new GitHubClient(_options.GitLocation, _options.GitHubPat, Logger, temporaryRepositoryPath: null, cache: null);
+
+            IRemote barRemote = RemoteFactory.GetBarOnlyRemote(_options, Logger);
+
+            var gitFileManager = new GitFileManager(targetBuildRepoClient, Logger);
```

Comment:
Once you change this around, you don't need to use the GitFileManager directly.  You can just use Remote.GetDependenciesAsync()